### PR TITLE
fix(provider/kubernetes): added logic to handle missing kind.

### PR DESCRIPTION
**Pull request by chlung**
About this PR,

- Added logic to handle server group operations which did not contain controller type.
- Fixed bugs found from smoke test.

Thanks. 
We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

**Comment by chlung** [comment]
@lwander please review. thanks.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 1]
```diff
@@ -361,5 +361,15 @@ class KubernetesClientApiAdapter {
       }
     }
   }
+
+  String getControlleType(String name, String namespace) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: Typo in "Controller"

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 2]
```diff
@@ -361,5 +361,15 @@ class KubernetesClientApiAdapter {
...
+    exceptionWrapper("getControlleType", "Get controller type ${name}", namespace) {
+      def deployedControllerSet = getStatefulSet(name, namespace)
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: What if it's a replica set, or a deployment, or a job?

Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: the call will ignore these controller type and continue.

**Comment by chlung** [comment]
@lwander  fix per your feedback.  please review.  thanks.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 3]
```diff
@@ -255,11 +256,11 @@ class KubernetesClientApiAdapter {
   V1beta1StatefulSet getStatefulSet(String statefulSetName, String namespace) {
     exceptionWrapper("statefulSets.create", "Get Stateful Set ${statefulSetName}", namespace) {
       try {
-        return apiInstance.readNamespacedStatefulSet(statefulSetName, namespace, API_CALL_RESULT_FORMAT, null, null)
+        apiInstance.readNamespacedStatefulSet(statefulSetName, namespace, API_CALL_RESULT_FORMAT, null, null)
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: Why omit return?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 4]
```diff
@@ -301,11 +302,11 @@ class KubernetesClientApiAdapter {
   V1beta1DaemonSet getDaemonSet(String name, String namespace) {
     exceptionWrapper("DaemonSet.get", "Get Daemon Set ${name}", namespace) {
       try {
-        return extApi.readNamespacedDaemonSet(name, namespace, API_CALL_RESULT_FORMAT, true, false)
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: Same question here as above

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 5]
```diff
@@ -361,5 +362,23 @@ class KubernetesClientApiAdapter {
       }
     }
   }
+
+  String getControllerType(String name, String namespace, String controllerType) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: Why pass in the type to get the type? I'm not clear on what this is supposed to do

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy [hunk 6]
```diff
@@ -964,7 +964,11 @@ class KubernetesClientApiConverter {
     return daemonset
   }
 
-  static boolean canUpdated(DeployKubernetesAtomicOperationDescription description) {
+  static Boolean canUpdated(DeployKubernetesAtomicOperationDescription description) {
     return description.updateController?.enabled
   }
+
+  static Boolean checkSequence(DeployKubernetesAtomicOperationDescription description) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy]: What does it mean to check the sequence? It seems to check that it's non negative. Can you update the function name?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy [hunk 7]
```diff
@@ -964,7 +964,11 @@ class KubernetesClientApiConverter {
     return daemonset
   }
 
-  static boolean canUpdated(DeployKubernetesAtomicOperationDescription description) {
+  static Boolean canUpdated(DeployKubernetesAtomicOperationDescription description) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy]: Can you name this more clearly? "canUpdated" isn't clear on what the intent is here.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy [hunk 8]
```diff
@@ -159,8 +161,12 @@ abstract class AbstractEnableDisableKubernetesAtomicOperation implements AtomicO
...
   }
 
-  Boolean supportsEnableDisable() {
-    switch(description.kind) {
+  boolean supportsEnableDisable(KubernetesV1Credentials credentials, String serverGroupName, String namespace) {
+    def controllerType = description.kind
+    if (!description.kind) {
+      controllerType = credentials.clientApiAdaptor.getControllerType(serverGroupName, namespace, null)
+    }
+    switch(controllerType) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy]: Missing space before (

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy [hunk 9]
```diff
@@ -66,6 +64,10 @@ abstract class AbstractEnableDisableKubernetesAtomicOperation implements AtomicO
 
     task.updateStatus basePhase, "Finding requisite server group..."
 
+    if (!supportsEnableDisable(credentials, description.serverGroupName, namespace)) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy]: This should probably log that the operation has completed early because enable/disable is unsupported.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 10]
```diff
@@ -97,28 +91,34 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
...
+    destroyController(credentials, namespace, description.serverGroupName, autoscalerName)
+
     task.updateStatus BASE_PHASE, "Successfully destroyed server group $description.serverGroupName."
   }
 
-  void destroyController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, String autoscalerName) {
-    switch(description.kind) {
+  void destroyController(KubernetesV1Credentials credentials, String namespace, String controllerName, String autoscalerName) {
+    def controllerType = description.kind
+    switch(controllerType) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: Missing space before (

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 11]
```diff
@@ -97,28 +91,34 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
...
 
+    destroyController(credentials, namespace, description.serverGroupName, autoscalerName)
+
     task.updateStatus BASE_PHASE, "Successfully destroyed server group $description.serverGroupName."
   }
 
-  void destroyController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, String autoscalerName) {
-    switch(description.kind) {
+  void destroyController(KubernetesV1Credentials credentials, String namespace, String controllerName, String autoscalerName) {
+    def controllerType = description.kind
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: This sounds like it should be `controllerKind`

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 12]
```diff
@@ -97,28 +91,34 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
...
-        credentials.apiClientAdaptor.hardDestroyStatefulSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyStatefulSet(controllerName, namespace, null, null, null)
         break
       case KubernetesUtil.CONTROLLERS_DAEMONSET_KIND:
-        credentials.apiClientAdaptor.hardDestroyDaemonSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyDaemonSet(controllerName, namespace, null, null, null)
         break
       default:
-        throw new KubernetesOperationException("Controller type $description.kind is not support.")
+        if (credentials.clientApiAdaptor.getControllerType(controllerName, namespace, KubernetesUtil.CONTROLLERS_STATEFULSET_KIND)) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: This can potentially delete multiple controller types now?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy [hunk 13]
```diff
@@ -110,17 +108,25 @@ class ResizeKubernetesAtomicOperation implements AtomicOperation<Void> {
     task.updateStatus BASE_PHASE, "Completed resize operation."
   }
 
-  void resizeController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, int size) {
-    if (description.kind == KubernetesUtil.CONTROLLERS_STATEFULSET_KIND) {
+  boolean resizeController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, int size) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: Why does this return a boolean of whether or not this was deleted? That doesn't seem related to this function.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy [hunk 14]
```diff
@@ -110,17 +108,25 @@ class ResizeKubernetesAtomicOperation implements AtomicOperation<Void> {
...
+
+    if (controllerType == KubernetesUtil.CONTROLLERS_STATEFULSET_KIND) {
       def deployedControllerSet = credentials.clientApiAdaptor.getStatefulSet(serverGroupName, namespace)
       if (deployedControllerSet) {
         credentials.apiClientAdaptor.resizeStatefulSet(serverGroupName, namespace, size)
       }
-    } else if (description.kind == KubernetesUtil.CONTROLLERS_DAEMONSET_KIND) {
+      isDeleted = true
+    } else if (controllerType == KubernetesUtil.CONTROLLERS_DAEMONSET_KIND) {
       throw new KubernetesOperationException("Not support resizing DaemoneSet.")
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: "Does not support resizing DaemonSet"

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy [hunk 15]
```diff
@@ -110,17 +108,25 @@ class ResizeKubernetesAtomicOperation implements AtomicOperation<Void> {
...
+      controllerType = credentials.clientApiAdaptor.getControllerType(serverGroupName, namespace)
+    }
+
+    if (controllerType == KubernetesUtil.CONTROLLERS_STATEFULSET_KIND) {
       def deployedControllerSet = credentials.clientApiAdaptor.getStatefulSet(serverGroupName, namespace)
       if (deployedControllerSet) {
         credentials.apiClientAdaptor.resizeStatefulSet(serverGroupName, namespace, size)
       }
-    } else if (description.kind == KubernetesUtil.CONTROLLERS_DAEMONSET_KIND) {
+      isDeleted = true
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: Why is this true here?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy (hunk 5 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: This mainly to deal with same for different controller type.  we need to go through each type.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy (hunk 7 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiConverter.groovy]: To ensure the updateController field is presented.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy (hunk 12 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: yes.  this  is found in smoke test.

**Comment by chlung** [comment]
@lwander  I updated code per your feedback.  Please review.  Thanks.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy (hunk 13 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: Ping

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy [hunk 16]
```diff
@@ -174,16 +173,22 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
     return replicaSet
   }
 
-  HasMetadata deployController(KubernetesV1Credentials credentials, String controllerName, String clusterName, String namespace) {
+  HasMetadata deployController(KubernetesV1Credentials credentials, def serverGroupNameResolver, String clusterName, String namespace) {
     def controllerSet
-    def canUpdated = KubernetesClientApiConverter.canUpdated(description)
+    def isUpdateControllerEnabled = KubernetesClientApiConverter.isUpdateControllerEnabled(description)
+    def controllerName
+    if (KubernetesClientApiConverter.validateSequence(description)) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: If validation fails, shouldn't that be a user error? e.g. the sequence desired can't be deployed?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy [hunk 17]
```diff
@@ -159,13 +162,19 @@ abstract class AbstractEnableDisableKubernetesAtomicOperation implements AtomicO
     null // Return nothing from void
   }
 
-  Boolean supportsEnableDisable() {
-    switch(description.kind) {
+  boolean supportsEnableDisable(KubernetesV1Credentials credentials, String serverGroupName, String namespace) {
+    def controllerKind = description.kind
+    if (!description.kind) {
+      controllerKind = credentials.clientApiAdaptor.getControllerKind(serverGroupName, namespace, null)
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy]: What if there is more than one controller with this name?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 18]
```diff
@@ -97,28 +91,34 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
...
-        credentials.apiClientAdaptor.hardDestroyStatefulSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyStatefulSet(controllerName, namespace, null, null, null)
         break
       case KubernetesUtil.CONTROLLERS_DAEMONSET_KIND:
-        credentials.apiClientAdaptor.hardDestroyDaemonSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyDaemonSet(controllerName, namespace, null, null, null)
         break
       default:
-        throw new KubernetesOperationException("Controller type $description.kind is not support.")
+        if (credentials.clientApiAdaptor.getControllerKind(controllerName, namespace, KubernetesUtil.CONTROLLERS_STATEFULSET_KIND)) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: Delete can now delete up to three server groups at once? I don't think this is a good practice

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy (hunk 13 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: I committed my changes, but this file hadn't been updated yet.  Let me find out what happen.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy (hunk 18 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: this is mainly to deal with smoke test framework which use one call to delete all controller.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy (hunk 16 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: Then, it should not use sequence which it will go to else.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy (hunk 17 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/AbstractEnableDisableKubernetesAtomicOperation.groovy]: I can add to default, but it is just print out the message.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy (hunk 13 above)
Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/ResizeKubernetesAtomicOperation.groovy]: now, code is update-to-date.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 19]
```diff
@@ -100,25 +99,26 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
...
         if (credentials.apiClientAdaptor.getAutoscaler(namespace, autoscalerName)) {
           task.updateStatus BASE_PHASE, "Destroying autoscaler..."
           if (!credentials.clientApiAdaptor.deleteAutoscaler(namespace, autoscalerName, null, null, null)) {
             throw new KubernetesOperationException("Failed to delete associated autoscaler $autoscalerName in $namespace.")
           }
         }
-        credentials.apiClientAdaptor.hardDestroyStatefulSet(serverGroupName, namespace, null, null, null)
-        break
+        credentials.apiClientAdaptor.hardDestroyStatefulSet(controllerName, namespace, null, null, null)
+        return true
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: Now when someone wants to delete a replica set, it might delete the stateful set instead - this doesn't feel like an improvement yet. If you need this functionality, please put it behind a feature flag so we can release.

**Comment by chlung** [comment]
Close this PR due to code push problem.  Open https://github.com/spinnaker/clouddriver/pull/2351.
